src/custom/state/fee/reducer.ts

```diff
@@ -0,0 +1,36 @@
+import { createReducer } from '@reduxjs/toolkit'
+import { updateFee, clearFee } from './actions'
+
+export interface FeeInformation {
+  expirationDate: string
+  minimalFee: string
+  feeRatio: number
+}
+
+interface FeeInformationObject {
+  token: string // token address
+  fee: FeeInformation
+}
+
+// {token address => FeeInformationObject} mapping
+type FeesMap = Record<string, FeeInformationObject>
+
+export interface FeeInformationState {
+  readonly feesMap: Partial<FeesMap>
+}
+
+const initialState: FeeInformationState = {
+  feesMap: {}
```

Comment:
Should we also differentiate by `chainId` or do we count on token addresses not overlapping over diff networks?

Reply:
hmm true, likely will. @anxolin thoughts

Reply:
token addresses do overlap in different networks